src/plugins/BehaviorSpecGenerator/BehaviorSpecGenerator.js

```diff
@@ -232,18 +234,29 @@ define([
             componentTypeNames = {},
             name,
             nodePath,
+            totalguardNames = {},
```

Comment:
For each componentType we are doing computation for children of the component.
So why do you want componentType seperately ?